docs/adr/0001-user-roles.rst

```diff
@@ -0,0 +1,51 @@
+ADR 0001 - User roles
+=====================
+
+Status
+------
+
+Proposed
+
+
+Context
+-------
+
+Depending on what we expect them to do on Marcha, we need to define some user roles.
+
+First we have the people managing a Marcha instance.
+
+Then we have people linking their own website (this website is a "consumer site") to a Marcha instance, to host videos.
+
+Next we have the video authors.
+
+And finally we have the users coming to watch videos.
+
+
+Decision
+--------
+
+For the first group of people, the ``is_superuser`` flag on the Django User model is enough.
+
+For the second group of people, we first need a Django model to host the link between the Marcha instance and the consumer. We name this model ``ConsumerSite``.
+
+This ``ConsumerSite`` model can have many admins (to avoid having no admin if the only one existing is not available anymore). So there is a ``ManyToMany`` link between this model and the ``User`` model. Having this link make the user an admin on the consumer site.
+
+It's not in the scope of this document to decide how the entries in the ``ConsumerSite`` model and the admins (and the links between both) are created.
+
+For the third group of people, the authors, we don't need them to create an account, but they still have one: when coming from the consumer site, to upload a video, the link/form to do so contains a token unique for this author (the same every time for the same user, or if not the same, it will still allow to identify the same user) that will create the account on Marcha the first time, and authenticate it every time. Like a SSO.
```

Comment:
:+1:

Reply:
Think "upload a video" or "choose from existing videos" and existing videos proposed are either mine or the ones from my organization if the organization manager gave access?
Corollary on models design is a Many2One from the author to the organization?

Reply:
Ok I didn't think about having to access previously uploaded videos, and even less about choosing a video from which I am not the author. It's a lot more complicated in this case. What is the idea? The manager can pick videos from any of its authors and say 'ok this one is available for all my authors' or 'this one is available to this author and this one only' ?

Reply:
A flag on the organization for the manager to control whether videos linked to courses of this organization are visible by all course authors or not. 

This because it will be the case when organizations are universities but not whe they are companies.

Reply:
Ok got it. Simple enough (not having to choose which authors can access is a lot simpler!!)